### PR TITLE
PDS + ScanCode

### PR DESCRIPTION
**Pull request by Jeeppler**
First version of PDS + ScanCode

- PDS + ScanCode container and scripts
- ScanCode can output multiple formats: SPDX Tag-Value, SPDX JSON, SPDX RDF, JSON and JSON Pretty Printed
- ScanCode uses [SPDX Java Tool](https://github.com/spdx/tools-java) to convert SPDX Tag-Value format into SPDX JSON as ScanCode is currently not able to produce SPDX JSON directly
- Helm chart to deploy PDS + ScanCode to Kubernetes
- Documentation

closes: #1030

### CODE REVIEW

File: sechub-pds-solutions/scancode/env-example [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# The PDS start mode
+#   debug - starts only the container
+#   localserver - starts the server locally
+PDS_START_MODE=debug, localserver
```
Comment (sven-dmlr) [inline sechub-pds-solutions/scancode/env-example:4]: Is this example syntactically correct?
I'd suggest to make it work and put only one (reasonable) option. E.g.:
`PDS_START_MODE=localserver`

(similar for the rest of the file)

File: sechub-pds-solutions/scancode/helm/pds-scancode/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<!-- SPDX-License-Identifier: MIT --->
+# ScanCode + PDS
+
+This Helm chart enables one to deploy [ScanCode](https://github.com/nexB/scancode-toolkit) a free and open source licence scannig tool and the [SecHub Product Delegation Server (PDS)](https://mercedes-benz.github.io/sechub/latest/sechub-product-delegation-server.html) into a Kubernetes environment. It is recommended to use the ScanCode + PDS in combination with [SecHub](https://mercedes-benz.github.io/sechub/).
```
Comment (sven-dmlr) [inline sechub-pds-solutions/scancode/helm/pds-scancode/README.md:4]: Maybe link to SecHub's main Github page like for ScanCode above?
```suggestion
This Helm chart enables one to deploy [ScanCode](https://github.com/nexB/scancode-toolkit) a free and open source licence scannig tool and the [SecHub Product Delegation Server (PDS)](https://mercedes-benz.github.io/sechub/latest/sechub-product-delegation-server.html) into a Kubernetes environment. It is recommended to use the ScanCode + PDS in combination with [SecHub](https://github.com/mercedes-benz/sechub).
```

**Review by sven-dmlr — CHANGES_REQUESTED** [review]
Good work! :+1:  Looking forward to use ScanCode.
Suggested a couple of small changes.

**Review by haerter-tss — APPROVED** [review]

File: sechub-pds-solutions/scancode/docker/ScanCode-Ubuntu.dockerfile [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+# Shared volumes
+ENV SHARED_VOLUMES="/shared_volumes"
+ENV SHARED_VOLUME_UPLOAD_DIR="$SHARED_VOLUMES/uploads"
+
+# Python version
+ARG PYTHON_VERSION="3.9"
+
+# Scancode
+ENV SCANCODE_VERSION="30.1.0"
+ARG SCANCODE_CHECKSUM="a9e43fdef934335e69e4abf77896225545d3e1fdbbd477ebabc37a4fa5ee2015  scancode-toolkit-30.1.0_py39-linux.tar.xz"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/ScanCode-Ubuntu.dockerfile:30]: I need the SCANCODE_VERSION later on. I don't need the SCANCODE_CHECKSUM except for the build. Scancode does not provide a checksum file. I created an issue for this in the ScanCode repo: https://github.com/nexB/scancode-toolkit/issues/2883.

File: sechub-pds-solutions/scancode/docker/ScanCode-Ubuntu.dockerfile [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+# SPDX Tool
+ENV SPDX_TOOL_VERISON="1.0.4"
+ARG SPDX_TOOL_CHECKSUM="e8da16d744d9a39dbc0420f776e0ebae71ce6a0be722941ada2e0e0f755cc4d0  tools-java-1.0.4-jar-with-dependencies.jar"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/ScanCode-Ubuntu.dockerfile:34]: I need the SPDX_TOOL_VERISONlater on. I don't need the SPDX_TOOL_CHECKSUMexcept for the build. SPDX does not provide a checksum file. I created an issue for this in the SPDX Tool repo: https://github.com/spdx/tools-java/issues/52.

File: sechub-pds-solutions/scancode/docker/ScanCode-Ubuntu.dockerfile [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+     && useradd --uid 2323 --no-log-init --create-home --gid pds pds
+
+# Create tool, pds, shared volume and download folder
+RUN  mkdir --parents "$TOOL_FOLDER" "$DOWNLOAD_FOLDER" "$PDS_FOLDER" "$SHARED_VOLUME_UPLOAD_DIR" "$WORKSPACE" && \
+    # Change owner and workspace and shared volumes folder
+    # the only two folders pds really needs write access to
+    chown --recursive pds:pds "$WORKSPACE" "$SHARED_VOLUMES"
+
+# Update image and install dependencies
+ENV DEBIAN_FRONTEND noninteractive
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/ScanCode-Ubuntu.dockerfile:48]: This is a good idea. For more information: https://bobcares.com/blog/debian_frontendnoninteractive-docker/

File: sechub-pds-solutions/scancode/env-example (hunk 1 above)
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/env-example:4]: The purpose of this file is to show all possible options and values. It is intended for documentation purposes only. This file does not have to be syntactically correct.

**Review by sven-dmlr — APPROVED** [review]

**Pull request merged**